pkg/lib/interaction/ratelimit.go

```diff
@@ -63,3 +64,22 @@ func AntiSpamSendOOBCodeBucket(oobType OOBType, target string) ratelimit.Bucket
 		ResetPeriod: duration.PerMinute,
 	}
 }
+
+type OTPCodeBucket struct {
+	Bucket ratelimit.Bucket
+}
+
+func ProvideOTPCodeBucket(cfg *config.OTPConfig) OTPCodeBucket {
+	return OTPCodeBucket{
+		Bucket: ratelimit.Bucket{
+			Size:        1,
+			ResetPeriod: cfg.SMS.ResendCooldownSeconds.Duration(),
+		},
+	}
+}
+
+func AntiSpamOTPCodeBucket(target string) OTPCodeBucket {
+	b := OTPCodeBucket{}
+	b.Bucket.Key = fmt.Sprintf("otp-code:%s", target)
+	return b
```

Comment:
This is incorrect. The Size and ResetPeriod of b is zero value. The correct way here is to introduce a struct that can return a ratelimit.Bucket.

```go
type AntiSpamOTPCodeBucketMaker struct {
  Config *config.OTPConfig // This field is dependency injected.
}

func (m *AntiSpamOTPCodeBucketMaker) MakeBucket(target string) ratelimit.Bucket {
  return ratelimit.Bucket{
    Key: fmt.Sprintf("otp-code:%s", target),
    Size: 1,
    ResetPeriod: m.Config.Blahblahblah,
  }
}
```

```go
// Other places that need the bucket inject the maker.

type AntiSpamOTPCodeBucketMaker interface {
  MakeBucket(target string) ratelimit.Bucket
}
```